crates/red_knot_python_semantic/src/types.rs

```diff
@@ -649,26 +651,35 @@ impl<'db> Type<'db> {
                         },
                     )
             }
-            (Type::ClassLiteral(..), Type::Instance(InstanceType { class }))
-                if class.is_known(db, KnownClass::Type) =>
-            {
-                true
-            }
```

Comment:
This is just a special case of the metaclass check below, which I've extended to work for `ClassLiteral(C)` as well as `SubclassOf(C)`